prompt/select.ts

```diff
@@ -99,6 +146,167 @@ export class Select<S extends SelectSettings = SelectSettings>
     return this.options[this.listIndex]?.value ?? this.settings.default;
   }
 
+  protected async submit(): Promise<void> {
+    const itemToSubmit = this.options[this.listIndex];
+
+    if (Select.itemIsBackButton(itemToSubmit)) {
+      const previousLevel = this.#parentCategories.pop();
+      if (typeof previousLevel === "object") {
+        this.options = previousLevel.options;
+        if (this.isCurrentlyInsideCategory()) {
+          this.listIndex = 2;
+        } else {
+          this.listIndex = 0;
+        }
+      }
+    } else if (Select.itemIsCategory(itemToSubmit)) {
+      this.#parentCategories.push({
+        options: this.options,
+        selectedCategoryIndex: this.listIndex,
+      });
+      this.options = this.itemsInsideCategory(itemToSubmit);
+      this.listIndex = 2;
+    } else {
+      await super.submit();
+    }
+  }
+
+  private itemsInsideCategory(
+    option: SelectOptionSettings,
+  ): SelectValueSettings {
+    return [
+      backItem,
+      ...Select.mapOptions([Select.separator(), ...option.options]),
+    ];
+  }
+
+  protected body(): string | Promise<string> {
+    let categoryText = "";
+
+    switch (this.#parentCategories.length) {
+      case 0:
+        return super.body();
+      case 1:
+        categoryText = dim(italic(this.nameOfParentCategory(0)));
+        break;
+
+      case 2: 
+        categoryText = dim(
+          italic(
+            this.nameOfParentCategory(0) + "/" + this.nameOfParentCategory(1),
+          ),
+        );
+        break;
+      
+      default: 
+        categoryText = dim(
+          italic(
+            this.nameOfParentCategory(0) + "/../" +
```

Comment:
I think i would use `›` instead of `/` for the breadcrumbs:
- `Foo › .. › Bar`
- or `› Foo › .. › Bar`

Reply:
And i think we can show more items by default and maybe make it configurable in form of max items or max length.

Reply:
I still don't know what the text styling should be though, to make sure it looks distinct enough from the actual prompting.

Reply:
I also changed the behavior a bit, so it's the first elements that get removed, instead of the middle ones. That was easier 😁

Reply:
ince i combined the back button and the separator, i think it's fine if the separator looks the same as other categories.